### PR TITLE
Fix/issue 281

### LINKED ISSUE #281: H5Store pytest is failing on MacOS but passes on Linux
### Description

Currently, one of the multiprocessing unit tests, `test_single_writer_multiple_reader_same_instance` from `test_h5store.py`, will fail on MacOS. The test passes on Linux and is skipped on Windows due to failure for an unknown reason. 

My system has `HDF5` version 10.4.1 and `h5py` version 2.10.0.

### To reproduce
On Mac:
```bash
$ cd /path/to/signac/
$ pytest tests/test_h5store.py -k "TestH5StoreMultiProcessing"
```
This will return an error. This test will pass on Linux (tested locally and tested in CI) and is skipped on Windows (not tested locally and tested on CI)

### Error output

```bash
OSError: Unable to open file (unable to lock file, errno = 35, error message = 'Resource temporarily unavailable')
```

### System configuration

Please complete the following information:

 - Operating System [e.g. macOS]: macOS 
 - Version of Python [e.g. 3.7]: 3.8.1
 - Version of signac [e.g. 1.0]: 1.3.0

Or copy & paste the output of: `python -c 'import platform; print(platform.platform()); import sys; print(sys.version); import signac; print(signac.__version__)'`

### PR DESCRIPTION
**Pull request by vyasr**
## Description

The test modified by this PR was reading from an hdf5 file in one process while writing to it in a separate process using nested contexts. Even a single concurrent read is not safe while writing in other processes, but I suspect that parallel-safe reads for a single reader might be an implementation detail in some versions of hdf5 but not others (leading to the intermittently successful behavior on OS X and the consistently correct behavior on Linux). Given that a number of the other tests explicitly test SWMR, however, I'm not sure why this test doesn't have it enabled. Perhaps @bdice or @klywang can shed some light on that, I believe they wrote/inspected/modified various versions of these tests.

## Motivation and Context

Resolves #281. In addition, the test will now pass on Windows.

## Types of Changes

- [ ] Documentation update
- [x] Bug fix
- [ ] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac/blob/master/changelog.txt) and added all related issue and pull request numbers for future reference (if applicable). See example below.

### CODE REVIEW

**Comment by vyasr** [comment]
@klywang since you weren't able to reproduce the problem anymore I'm not sure if you can validate these changes, but if you have any way to do so that would be super helpful!

File: tests/test_h5store.py [hunk 1]
```diff
@@ -690,23 +683,24 @@ def test_single_writer_multiple_reader_same_process(self):
...
+            with H5Store(self._fn_store, mode="r", swmr=True) as h5s:
+                list(h5s)
+
         def read():
-            p = Process(
-                target=_read_from_h5store,
-                args=(self._fn_store,),
-                kwargs=(dict(mode="r")),
-            )
+            p = Process()
```
Comment (bdice) [inline tests/test_h5store.py:696]: I'm confused. Does this process _do_ anything? It seems like it just gets created and joined.

**Comment by bdice** [comment]
@vyasr (tagging @glotzerlab/signac-committers for visibility and to solicit any contradicting opinions)

I believe I wrote these tests of SWMR behavior, but I've never felt confident about them and we've had a great deal of problems with them over time. Given how much effort we've expended on this sub-feature of HDF5 SWMR support (which to the best of my knowledge is not utilized at all by signac users), I'm not sure that it is in still in scope for us as signac maintainers to test extensive guarantees about SWMR behavior in signac. It seems so highly dependent on parts of the software stack that we don't control (specifics of the HDF5 build, h5py, Python implementation, and OS level support?!). Thus:

1. I am of the opinion that power users with extremely specific needs for SWMR are likely experienced enough to learn and correctly use h5py directly, rather than the simple wrapper that signac provides.
2. I am in favor of _literally any solution that will eliminate the problems SWMR has caused us in testing_, whether that means fixing tests so that they reliably pass, marking them as expected failures, or removing the tests entirely and advocating that users use h5py directly if they have such advanced needs.

File: tests/test_h5store.py (hunk 1 above)
Comment (vyasr) [reply tests/test_h5store.py]: Well that's strange. I'm not sure how it got pushed in this weird intermediate state, my fault for context switching between multiple branches. I've pushed the actual fix now.

**Comment by vyasr** [comment]
> @vyasr (tagging @glotzerlab/signac-committers for visibility and to solicit any contradicting opinions)
> 
> I believe I wrote these tests of SWMR behavior, but I've never felt confident about them and we've had a great deal of problems with them over time. Given how much effort we've expended on this sub-feature of HDF5 SWMR support (which to the best of my knowledge is not utilized at all by signac users), I'm not sure that it is in still in scope for us as signac maintainers to test extensive guarantees about SWMR behavior in signac. It seems so highly dependent on parts of the software stack that we don't control (specifics of the HDF5 build, h5py, Python implementation, and OS level support?!). Thus:
> 
>     1. I am of the opinion that power users with extremely specific needs for SWMR are likely experienced enough to learn and correctly use h5py directly, rather than the simple wrapper that signac provides.
> 
>     2. I am in favor of _literally any solution that will eliminate the problems SWMR has caused us in testing_, whether that means fixing tests so that they reliably pass, marking them as expected failures, or removing the tests entirely and advocating that users use h5py directly if they have such advanced needs.

I agree that SWMR is highly dependent on a lots of software that we don't control, and I'd be fine not worrying about it. We're not actually doing anything to support SWMR; any user who wants SWMR support is expected to pass in the appropriate kwargs to transparently be passed through to h5py. Having said that, while I wasn't very involved with earlier passes through this code so I don't know what caused the problems with tests, at least in this case the issue is an improper usage of h5py in the test itself. I'd say that we move forward with this fix for now since there haven't been any recent hdf5 issues that I'm aware of aside from this one (even this issue is almost a year old at this point). If we run into more problems in the future I'd be fine revisiting the question of just removing these tests since ultimately they're largely testing external code.

**Review by bdice — APPROVED** [review]
Looks good to me.

**Pull request merged**